### ISSUE TITLE
[FEATURE] `no_push: true` should not complain about missing credentials

### ISSUE THREAD

**Issue by casperdcl** [elgohr/Publish-Docker-Github-Action#110]
When `no_push: true`, don't complain about blank username/password.

Use case:

```yaml
- uses: elgohr/Publish-Docker-Github-Action@master
  with:
    name: ${{ github.repository }}
    password: ${{ secrets.DOCKER_PWD }}  # obviously blank on pull_request
    username: ${{ secrets.DOCKER_USR }}  # obviously blank on pull_request
    no_push: ${{ github.event_name == 'pull_request' }}
```

this will allow testing that the image builds on pull requests, but won't try to login. (Neater than duplicating two different versions of the block with `if: github.event_name == 'pull_request'` and `github.event_name != 'pull_request'`.)

**Comment by elgohr** [comment]
Hey,
totally understand why you come to that point.
Sadly the login is necessary, due to the fact that you could depend on an image from a private registry.
Makes sense?

**Comment by casperdcl** [comment]
yes but that means the login should be optional. As you say, it's only necessary for builds which require login to private registries.

Your argument is like saying github should not allow pulling public repos without login.

**Comment by casperdcl** [comment]
will change the PR slightly to allow blank login iff `no_push` is defined.

**Comment by elgohr** [comment]
So how would you know whether the image need credentials for download or not inside the action?

**Comment by casperdcl** [comment]
already updated #111 3 days ago btw...

credentials|required|not required
-|-|-
**provided**|ok|ok
**not provided**|fails|should not fail; fixed by #111

**Comment by elgohr** [comment]
Please give me some time, will have a look
